scripts/contacts/migrateContactsV1toV2.js

```diff
@@ -0,0 +1,167 @@
+    return KONNECTOR_APP_NAME
+  } else if (metadataKeys.includes('cozy')) {
+    return CONTACTS_APP_NAME
+  } else if (metadataKeys.length === 0) {
+    return VCARD_APP_NAME
+  }
+
+  return null
+}
+
+const migrateContactV1toV2 = (contact, accountId, contactAccountId) => {
+  const now = new Date().toISOString()
+  const createdByApp = getCreatedByApp(contact)
+  let accounts = []
+  let sourceAccount, sync
+  if (contactAccountId && createdByApp === KONNECTOR_APP_NAME) {
+    sync = {
+      [contactAccountId]: {
+        contactsAccountsId: contactAccountId,
+        id: contact.vendorId,
+        konnector: KONNECTOR_APP_NAME,
+        lastSync: now,
+        remoteRev: null
+      }
+    }
+    sourceAccount = accountId
+    accounts = [
+      {
+        _id: contactAccountId,
+        _type: DOCTYPE_CONTACTS_ACCOUNT
+      }
+    ]
+  }
+
+  return {
+    ...contact,
+    me: false,
+    cozyMetadata: {
+      sync,
+      createdAt: null,
+      createdByApp,
+      doctypeVersion: DOCTYPE_CONTACTS_VERSION,
+      sourceAccount,
+      updatedAt: now,
+      updatedByApps: [createdByApp]
+    },
+    relationships: {
+      ...contact.relationships,
+      accounts: {
+        data: accounts
+      }
+    }
+  }
+}
+
+const migrateContactsV1toV2 = (contacts, accountId, contactAccountId) =>
+  contacts.map(contact =>
+    migrateContactV1toV2(contact, accountId, contactAccountId)
+  )
+
+const createContactAccount = (account, accountName) => ({
+  canLinkContacts: true,
+  shouldSyncOrphan: true,
+  lastSync: null,
+  lastLocalSync: null,
+  name: accountName,
+  type: KONNECTOR_APP_NAME,
+  sourceAccount: account._id,
+  version: ACCOUNT_APP_VERSION
+})
+
+const getAccountName = contacts => {
+  const contactWithFrom = contacts.find(contact =>
+    has(contact, 'metadata.google.from')
+  )
+  return contactWithFrom ? contactWithFrom.metadata.google.from : ''
+}
+
+async function doMigrations(dryRun) {
+  const logWithInstance = getWithInstanceLogger(client)
+  const accounts = await api.fetchAll(DOCTYPE_ACCOUNTS)
+  const contacts = await api.fetchAll(DOCTYPE_CONTACTS)
+  const konnectorAccount = accounts.find(doc => doc.account_type === 'google')
+  const accountName = getAccountName(contacts)
+
+  let accountId, contactAccountId
+  if (dryRun) {
+    accountId = 'dryRunAccountId'
+    contactAccountId = 'dryRunContactAccountId'
+  } else if (konnectorAccount) {
+    const contactAccount = createContactAccount(konnectorAccount, accountName)
+    const response = await api.updateAll(DOCTYPE_CONTACTS_ACCOUNT, [
+      contactAccount
+    ])
+    accountId = konnectorAccount._id
+    contactAccountId = response[0].id
+  }
+
+  const updatedContacts = migrateContactsV1toV2(
+    contacts,
+    accountId,
+    contactAccountId
+  )
+
+  if (!dryRun) {
+    await api.updateAll(DOCTYPE_CONTACTS, updatedContacts)
+  } else {
+    logWithInstance(
+      'Dry run: first updated contact\n',
+      diff(contacts[0], updatedContacts[0])
```

Comment:
Why not log all updates?

Reply:
I think it would take too much time if we display everything. The aim of the dry run is to ensure there is no error during data conversion (and to be honest, I was inspired by banks migration)